activedata_etl/transforms/jscov_to_es.py

```diff
@@ -63,11 +67,37 @@ def generator():
                         source=obj.get("sourceFile"),
                         cause=e
                     )
+    with TempDirectory() as tmpdir:
+        Log.note('Using temp dir: {{dir}}', dir=tmpdir)
+        jsdcov_file = File.new_instance(tmpdir, "jsdcov.zip").abspath
+
+        download_file(artifact.url, jsdcov_file)
+        unzip_dir = File.new_instance(tmpdir, "jsdcov").abspath
+        unzip_files(jsdcov_file, unzip_dir)
+
+        key = etl2key(artifact_etl)
+        Log.note('Writing artifacts to destination. ')
+        for filename in listdir(unzip_dir):
+            file_dir = path.join(unzip_dir, filename)
+            destination.write_lines(key, (value2json(d) for d in generator(file_dir)))
+        keys = [key]
+        return keys
+
+
+def download_file(url, destination):
+    Log.note('Fetching JSDCov artifact: {{url}}', url=url)
+    tempfile = file(destination, "w+b")
+    stream = http.get(url).raw
+    try:
+        for b in iter(lambda: stream.read(8192), b""):
+            tempfile.write(b)
+    finally:
+        stream.close()
+
 
-    key = etl2key(artifact_etl)
-    destination.write_lines(key, (value2json(d) for d in generator()))
-    keys = [key]
-    return keys
+def unzip_files(jsdcov_file, dest_dir):
```

Comment:
instead of unzipping to disk, read out of the zip file directly. The `grcov_to_es` has an example of this:

                with ZipFile(tmpfile.abspath) as zipped:
                    for num, zip_name in enumerate(zipped.namelist()):
                        if num == 1:
                            Log.error("expecting only one artifact in the grcov.zip file while processing {{key}}", key=source_key)
                        for source in parse_lcov_coverage(source_key, grcov_artifact.url, ibytes2ilines(zipped.open(zip_name))):